## 라이브러리 설치

In [1]:
pip install pymysql

Defaulting to user installation because normal site-packages is not writeable
Note: you may need to restart the kernel to use updated packages.


------------------

## Keydev

In [2]:
import pymysql
from sqlalchemy import create_engine
import pandas as pd

user = 'felab'
password = 'rlawkdgh'
host = '192.168.0.2'  # 포트 번호를 여기에 포함하지 않습니다.
port = 3306  # 포트 번호를 별도의 변수로 설정합니다.
database = 'db'
table_name = 'keydev' #읽고 싶은 데이터셋.

# SQLAlchemy 엔진 생성 시 호스트와 포트를 정확하게 지정합니다.
engine = create_engine(f"mysql+pymysql://{user}:{password}@{host}:{port}/{database}")

# pymysql 연결 시에도 마찬가지로 호스트와 포트를 정확하게 지정합니다.
connection = pymysql.connect(host=host, port=port, user=user, password=password, database=database, connect_timeout = 10)

# 모든 컬럼 불러오기
query = f"SELECT * FROM {table_name}" # 원하는 쿼리. SQL 지식 필요함

# 원하는 컬럼 불러오기
# query = f"SELECT keyDevEventTypeName FROM {table_name}" # 원하는 쿼리. SQL 지식 필요함
keydev = pd.read_sql(query, engine)
keydev

companyId                           companyName      keyDevId  \
0        1.850100e+04                  21 Invest Sgr S.p.A.  1.786392e+09   
1        1.850100e+04                  21 Invest Sgr S.p.A.  1.775873e+09   
2        1.851100e+04                          3i Group plc  1.856147e+09   
3        1.851100e+04                          3i Group plc  1.846370e+09   
4        1.851100e+04                          3i Group plc  1.843429e+09   
...               ...                                   ...           ...   
3244073  2.147481e+09  Centennial Emergency Physicians, LLC  1.846943e+09   
3244074  2.147481e+09  Centennial Emergency Physicians, LLC  1.846939e+09   
3244075  2.147481e+09  Centennial Emergency Physicians, LLC  1.838307e+09   
3244076  2.147481e+09  Centennial Emergency Physicians, LLC  1.837743e+09   
3244077  2.147481e+09  Centennial Emergency Physicians, LLC  1.838307e+09   

                                                  headline  \
0        A&M Capital Europe acquired Carton Pack Srl fr...   
1        A&M Capital Europe agreed to acquire Carton Pa...   
2                              3i Group plc - Special Call   
3                           3i Group plc Approves Dividend   
4        3i Group plc to Report Q1, 2024 Results on Jul...   
...                                                    ...   
3244073  Joint Reorganization Plan and Disclosure State...   
3244074  Motion for De Minimis Asset Sale Filed by Envi...   
3244075  Motion for Joint Administration Approved for E...   
3244076  Envision Healthcare Corporation Filed for Bank...   
3244077  Motion for Joint Administration Filed by Envis...   

                                                 situation  \
0        A&M Capital Europe agreed to acquire Carton Pa...   
1        A&M Capital Europe agreed to acquire Carton Pa...   
2                              3i Group plc - Special Call   
3        3i Group plc at its Annual General Meeting hel...   
4        3i Group plc announced that they will report Q...   
...                                                    ...   
3244073  Envision Healthcare Corporation, along with it...   
3244074  Envision Healthcare Corporation filed a motion...   
3244075  The US Bankruptcy Court granted an order for t...   
3244076  Envision Healthcare Corporation, along with it...   
3244077  Envision Healthcare Corporation, along with it...   

              announcedDate mostImportantDateUTC  \
0       2022-04-21 00:00:00  2022-04-21 00:00:00   
1       2022-03-11 00:00:00  2022-03-11 00:00:00   
2       2023-09-20 06:00:00  2023-09-20 13:00:00   
3       2023-06-29 14:30:00  2023-06-29 14:30:00   
4       2023-06-16 00:00:00  2023-07-20 00:00:00   
...                     ...                  ...   
3244073 2023-06-29 00:00:00  2023-06-29 04:00:00   
3244074 2023-05-31 00:00:00  2023-05-31 04:00:00   
3244075 2023-05-15 00:00:00  2023-05-15 04:00:00   
3244076 2023-05-15 00:00:00  2023-05-15 04:00:00   
3244077 2023-05-15 00:00:00  2023-05-15 04:00:00   

                         keyDevEventTypeName  
0                   M&A Transaction Closings  
1              M&A Transaction Announcements  
2                              Special Calls  
3                         Dividend Increases  
4                      Earnings Release Date  
...                                      ...  
3244073         Bankruptcy – Reorganization   
3244074  Bankruptcy – Asset Sale/Liquidation  
3244075                   Bankruptcy - Other  
3244076                  Bankruptcy - Filing  
3244077                   Bankruptcy - Other  

[3244078 rows x 8 columns]

## Company (행 7천개밖에 없음)

In [3]:
import pymysql
from sqlalchemy import create_engine
import pandas as pd

user = 'felab'
password = 'rlawkdgh'
host = '192.168.0.2'  # 포트 번호를 여기에 포함하지 않습니다.
port = 3306  # 포트 번호를 별도의 변수로 설정합니다.
database = 'db'
table_name = 'company' #읽고 싶은 데이터셋.

# SQLAlchemy 엔진 생성 시 호스트와 포트를 정확하게 지정합니다.
engine = create_engine(f"mysql+pymysql://{user}:{password}@{host}:{port}/{database}")

# pymysql 연결 시에도 마찬가지로 호스트와 포트를 정확하게 지정합니다.
connection = pymysql.connect(host=host, port=port, user=user, password=password, database=database)

query = f"SELECT * FROM {table_name}" # 원하는 쿼리. SQL 지식 필요함
company = pd.read_sql(query, engine)
company

companyId                             companyName  \
0       18493.0                     DC Venture Partners   
1       18495.0                 13i Capital Corporation   
2       18499.0          21 International Holdings Inc.   
3       18501.0                    21 Invest Sgr S.p.A.   
4       18505.0              Palo Alto Venture Partners   
...         ...                                     ...   
7140    29969.0     Insight Communications Company Inc.   
7141    29970.0                     Insight Development   
7142    29971.0  InSight Health Services Holdings Corp.   
7143    29972.0                      InsightExpress LLC   
7144    29973.0          Insignia Financial Group, Inc.   

                     streetAddress        country  officePhoneValue  \
0        1103 Riverbend Club Drive  United States      770-612-9190   
1                2021 Midwest Road  United States      630-620-2255   
2                             None  United States              None   
3           Viale G. Felissent, 90          Italy  39 0422 31 66 11   
4                555 Bryant Street  United States      650-462-1221   
...                            ...            ...               ...   
7140            810 Seventh Avenue  United States      917 286 2300   
7141             2420 Camino Ramon  United States      925-244-2000   
7142        26250 Enterprise Court  United States      949-282-6000   
7143             333 Ludlow Street  United States      203-359-4174   
7144  One Insignia Financial Plaza  United States      864 239 1000   

        officeFaxValue                 webPage tickerSymbol  yearFounded  \
0                 None                    None         None       1998.0   
1         800-849-4183      www.13icapital.com         None       1993.0   
2                 None                    None         None          NaN   
3     39 0422 31 66 00  www.21investimenti.com         None       1992.0   
4         650-462-1227                    None         None       1996.0   
...                ...                     ...          ...          ...   
7140      917 286 2301                    None         ICCI       1985.0   
7141      925-244-2020      www.insightdev.com         None          NaN   
7142      949-452-0161                    None       ISGT.Q       1982.0   
7143      203-359-4718  www.insightexpress.com         None       1999.0   
7144              None                    None          IFS       1990.0   

                   simpleIndustryDescription          companyTypeName  \
0                            Capital Markets  Private Investment Firm   
1                            Capital Markets  Private Investment Firm   
2                            Capital Markets  Private Investment Firm   
3                            Capital Markets  Private Investment Firm   
4                            Capital Markets  Private Investment Firm   
...                                      ...                      ...   
7140                                   Media          Private Company   
7141                                Software          Private Company   
7142      Health Care Providers and Services          Private Company   
7143                                   Media          Private Company   
7144  Real Estate Management and Development          Private Company   

                                longBusinessDescription  \
0                                                  None   
1                                                  None   
2                                                  None   
3                                                  None   
4                                                  None   
...                                                 ...   
7140  Insight Communications Company, Inc. operates ...   
7141                                               None   
7142  InSight Health Services Holdings Corp. provide...   
7143                                               None   
7144 

## Market (미업로드)

In [6]:
import pymysql
from sqlalchemy import create_engine
import pandas as pd

user = 'felab'
password = 'rlawkdgh'
host = '192.168.0.2'  # 포트 번호를 여기에 포함하지 않습니다.
port = 3306  # 포트 번호를 별도의 변수로 설정합니다.
database = 'db'
table_name = 'market' #읽고 싶은 데이터셋.

# SQLAlchemy 엔진 생성 시 호스트와 포트를 정확하게 지정합니다.
engine = create_engine(f"mysql+pymysql://{user}:{password}@{host}:{port}/{database}")

# pymysql 연결 시에도 마찬가지로 호스트와 포트를 정확하게 지정합니다.
connection = pymysql.connect(host=host, port=port, user=user, password=password, database=database)

query = f"SELECT * FROM {table_name}" # 원하는 쿼리. SQL 지식 필요함
market = pd.read_sql(query, engine)
market

Empty DataFrame
Columns: [companyId, companyName, tickerSymbol, exchangeSymbol, ISOCode, pricingDate, exDate, payDate, recordDate, announcedDate, priceClose, volume, adjustmentFactor, VWAP, divAdjFactor, marketCap, TEV, sharesOutstanding, divFreqTypeName, supplementalTypeName, dividendTypeName, divAmount, netAmount, grossAmount]
Index: []

[0 rows x 24 columns]

## Estimates

In [5]:
import pymysql
from sqlalchemy import create_engine
import pandas as pd

user = 'felab'
password = 'rlawkdgh'
host = '192.168.0.2'  # 포트 번호를 여기에 포함하지 않습니다.
port = 3306  # 포트 번호를 별도의 변수로 설정합니다.
database = 'db'
table_name = 'estimates' #읽고 싶은 데이터셋.

# SQLAlchemy 엔진 생성 시 호스트와 포트를 정확하게 지정합니다.
engine = create_engine(f"mysql+pymysql://{user}:{password}@{host}:{port}/{database}")

# pymysql 연결 시에도 마찬가지로 호스트와 포트를 정확하게 지정합니다.
connection = pymysql.connect(host=host, port=port, user=user, password=password, database=database)

query = f"SELECT * FROM {table_name}" # 원하는 쿼리. SQL 지식 필요함
estimates = pd.read_sql(query, engine)
estimates

companyId         companyName  fiscalQuarter  fiscalYear  \
0        1.851100e+04        3i Group plc            0.0         0.0   
1        1.851100e+04        3i Group plc            4.0      2021.0   
2        1.851100e+04        3i Group plc            4.0      2025.0   
3        1.851100e+04        3i Group plc            4.0      2024.0   
4        1.851100e+04        3i Group plc            4.0      2022.0   
...               ...                 ...            ...         ...   
2121823  1.851035e+09  Liberty Live Group            0.0         0.0   
2121824  1.851035e+09  Liberty Live Group            0.0         0.0   
2121825  1.851035e+09  Liberty Live Group            0.0         0.0   
2121826  1.851035e+09  Liberty Live Group            0.0         0.0   
2121827  1.851035e+09  Liberty Live Group            0.0         0.0   

        periodEndDate         advanceDate       effectiveDate     toDate  \
0                 NaT                 NaT 2023-09-21 17:28:40 2079-06-06   
1          2021-03-31 2021-05-13 06:00:00 2021-03-26 09:41:04 2079-06-06   
2          2025-03-31                 NaT 2023-07-24 05:16:06 2079-06-06   
3          2024-03-31                 NaT 2023-07-24 05:16:06 2079-06-06   
4          2022-03-31 2022-05-12 06:00:00 2022-05-09 22:10:06 2079-06-06   
...               ...                 ...                 ...        ...   
2121823           NaT                 NaT 2023-08-28 10:10:34 2079-06-06   
2121824           NaT                 NaT 2023-08-28 10:10:28 2079-06-06   
2121825           NaT                 NaT 2023-08-28 10:10:34 2079-06-06   
2121826           NaT                 NaT 2023-08-28 10:10:28 2079-06-06   
2121827           NaT                 NaT 2023-08-28 10:10:34 2079-06-06   

        periodTypeName  dataItemId                   dataItemName ISOCode  \
0         Non-Periodic    100160.0       Consensus Recommendation    None   
1               Annual    100191.0        EBITDA - # of Estimates    None   
2               Annual    100191.0        EBITDA - # of Estimates    None   
3               Annual    100191.0        EBITDA - # of Estimates    None   
4               Annual    100191.0        EBITDA - # of Estimates    None   
...                ...         ...                            ...     ...   
2121823   Non-Periodic    100164.0     Target Price Consensus Low     USD   
2121824   Non-Periodic    100161.0    Target Price Consensus Mean     USD   
2121825   Non-Periodic    100161.0    Target Price Consensus Mean     USD   
2121826   Non-Periodic    100162.0  Target Price Consensus Median     USD   
2121827   Non-Periodic    100162.0  Target Price Consensus Median     USD   

         dataItemValue estimateScaleName  
0                 1.75          Absolute  
1                 3.00          Absolute  
2                 2.00          Absolute  
3                 2.00          Absolute  
4                 1.00          Absolute  
...                ...               ...  
2121823          56.00          Absolute  
2121824          56.00          Absolute  
2121825          56.00          Absolute  
2121826          56.00          Absolute  
2121827          56.00          Absolute  

[2121828 rows x 14 columns]

## Exchangerate

In [7]:
import pymysql
from sqlalchemy import create_engine
import pandas as pd

user = 'felab'
password = 'rlawkdgh'
host = '192.168.0.2'  # 포트 번호를 여기에 포함하지 않습니다.
port = 3306  # 포트 번호를 별도의 변수로 설정합니다.
database = 'db'
table_name = 'exchangerate' #읽고 싶은 데이터셋.

# SQLAlchemy 엔진 생성 시 호스트와 포트를 정확하게 지정합니다.
engine = create_engine(f"mysql+pymysql://{user}:{password}@{host}:{port}/{database}")

# pymysql 연결 시에도 마찬가지로 호스트와 포트를 정확하게 지정합니다.
connection = pymysql.connect(host=host, port=port, user=user, password=password, database=database)

query = f"SELECT * FROM {table_name}" # 원하는 쿼리. SQL 지식 필요함
exchangerate = pd.read_sql(query, engine)
exchangerate

ISOCode  priceDate    priceClose
0          PTE 2021-03-20    178.318715
1          ROL 2021-03-20  25789.000000
2          SDD 2021-03-20    203.090000
3          SIT 2021-03-20    213.147798
4          SKK 2021-03-20     26.795571
...        ...        ...           ...
494831     BZD 2018-08-17      2.015300
494832     BZD 2018-08-18      2.015300
494833     BZD 2018-08-19      2.015300
494834     BZD 2018-08-20      2.015300
494835     BZD 2018-08-21      2.015300

[494836 rows x 3 columns]

## Financials  (미업로드)

--------------------

In [9]:
keydev

companyId                           companyName      keyDevId  \
0        1.850100e+04                  21 Invest Sgr S.p.A.  1.786392e+09   
1        1.850100e+04                  21 Invest Sgr S.p.A.  1.775873e+09   
2        1.851100e+04                          3i Group plc  1.856147e+09   
3        1.851100e+04                          3i Group plc  1.846370e+09   
4        1.851100e+04                          3i Group plc  1.843429e+09   
...               ...                                   ...           ...   
3244073  2.147481e+09  Centennial Emergency Physicians, LLC  1.846943e+09   
3244074  2.147481e+09  Centennial Emergency Physicians, LLC  1.846939e+09   
3244075  2.147481e+09  Centennial Emergency Physicians, LLC  1.838307e+09   
3244076  2.147481e+09  Centennial Emergency Physicians, LLC  1.837743e+09   
3244077  2.147481e+09  Centennial Emergency Physicians, LLC  1.838307e+09   

                                                  headline  \
0        A&M Capital Europe acquired Carton Pack Srl fr...   
1        A&M Capital Europe agreed to acquire Carton Pa...   
2                              3i Group plc - Special Call   
3                           3i Group plc Approves Dividend   
4        3i Group plc to Report Q1, 2024 Results on Jul...   
...                                                    ...   
3244073  Joint Reorganization Plan and Disclosure State...   
3244074  Motion for De Minimis Asset Sale Filed by Envi...   
3244075  Motion for Joint Administration Approved for E...   
3244076  Envision Healthcare Corporation Filed for Bank...   
3244077  Motion for Joint Administration Filed by Envis...   

                                                 situation  \
0        A&M Capital Europe agreed to acquire Carton Pa...   
1        A&M Capital Europe agreed to acquire Carton Pa...   
2                              3i Group plc - Special Call   
3        3i Group plc at its Annual General Meeting hel...   
4        3i Group plc announced that they will report Q...   
...                                                    ...   
3244073  Envision Healthcare Corporation, along with it...   
3244074  Envision Healthcare Corporation filed a motion...   
3244075  The US Bankruptcy Court granted an order for t...   
3244076  Envision Healthcare Corporation, along with it...   
3244077  Envision Healthcare Corporation, along with it...   

              announcedDate mostImportantDateUTC  \
0       2022-04-21 00:00:00  2022-04-21 00:00:00   
1       2022-03-11 00:00:00  2022-03-11 00:00:00   
2       2023-09-20 06:00:00  2023-09-20 13:00:00   
3       2023-06-29 14:30:00  2023-06-29 14:30:00   
4       2023-06-16 00:00:00  2023-07-20 00:00:00   
...                     ...                  ...   
3244073 2023-06-29 00:00:00  2023-06-29 04:00:00   
3244074 2023-05-31 00:00:00  2023-05-31 04:00:00   
3244075 2023-05-15 00:00:00  2023-05-15 04:00:00   
3244076 2023-05-15 00:00:00  2023-05-15 04:00:00   
3244077 2023-05-15 00:00:00  2023-05-15 04:00:00   

                         keyDevEventTypeName  
0                   M&A Transaction Closings  
1              M&A Transaction Announcements  
2                              Special Calls  
3                         Dividend Increases  
4                      Earnings Release Date  
...                                      ...  
3244073         Bankruptcy – Reorganization   
3244074  Bankruptcy – Asset Sale/Liquidation  
3244075                   Bankruptcy - Other  
3244076                  Bankruptcy - Filing  
3244077                   Bankruptcy - Other  

[3244078 rows x 8 columns]

In [10]:
keydev.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3244078 entries, 0 to 3244077
Data columns (total 8 columns):
 #   Column                Dtype         
---  ------                -----         
 0   companyId             float64       
 1   companyName           object        
 2   keyDevId              float64       
 3   headline              object        
 4   situation             object        
 5   announcedDate         datetime64[ns]
 6   mostImportantDateUTC  datetime64[ns]
 7   keyDevEventTypeName   object        
dtypes: datetime64[ns](2), float64(2), object(4)
memory usage: 198.0+ MB


In [11]:
# 각 컬럼별로 null값의 갯수 확인
keydev.isnull().sum()

companyId               0
companyName             0
keyDevId                0
headline                0
situation               4
announcedDate           0
mostImportantDateUTC    0
keyDevEventTypeName     0
dtype: int64

In [12]:
# 원본 데이터프레임에서 null값을 포함한 행 삭제
keydev.dropna(inplace=True)  

In [13]:
keydev

companyId                           companyName      keyDevId  \
0        1.850100e+04                  21 Invest Sgr S.p.A.  1.786392e+09   
1        1.850100e+04                  21 Invest Sgr S.p.A.  1.775873e+09   
2        1.851100e+04                          3i Group plc  1.856147e+09   
3        1.851100e+04                          3i Group plc  1.846370e+09   
4        1.851100e+04                          3i Group plc  1.843429e+09   
...               ...                                   ...           ...   
3244073  2.147481e+09  Centennial Emergency Physicians, LLC  1.846943e+09   
3244074  2.147481e+09  Centennial Emergency Physicians, LLC  1.846939e+09   
3244075  2.147481e+09  Centennial Emergency Physicians, LLC  1.838307e+09   
3244076  2.147481e+09  Centennial Emergency Physicians, LLC  1.837743e+09   
3244077  2.147481e+09  Centennial Emergency Physicians, LLC  1.838307e+09   

                                                  headline  \
0        A&M Capital Europe acquired Carton Pack Srl fr...   
1        A&M Capital Europe agreed to acquire Carton Pa...   
2                              3i Group plc - Special Call   
3                           3i Group plc Approves Dividend   
4        3i Group plc to Report Q1, 2024 Results on Jul...   
...                                                    ...   
3244073  Joint Reorganization Plan and Disclosure State...   
3244074  Motion for De Minimis Asset Sale Filed by Envi...   
3244075  Motion for Joint Administration Approved for E...   
3244076  Envision Healthcare Corporation Filed for Bank...   
3244077  Motion for Joint Administration Filed by Envis...   

                                                 situation  \
0        A&M Capital Europe agreed to acquire Carton Pa...   
1        A&M Capital Europe agreed to acquire Carton Pa...   
2                              3i Group plc - Special Call   
3        3i Group plc at its Annual General Meeting hel...   
4        3i Group plc announced that they will report Q...   
...                                                    ...   
3244073  Envision Healthcare Corporation, along with it...   
3244074  Envision Healthcare Corporation filed a motion...   
3244075  The US Bankruptcy Court granted an order for t...   
3244076  Envision Healthcare Corporation, along with it...   
3244077  Envision Healthcare Corporation, along with it...   

              announcedDate mostImportantDateUTC  \
0       2022-04-21 00:00:00  2022-04-21 00:00:00   
1       2022-03-11 00:00:00  2022-03-11 00:00:00   
2       2023-09-20 06:00:00  2023-09-20 13:00:00   
3       2023-06-29 14:30:00  2023-06-29 14:30:00   
4       2023-06-16 00:00:00  2023-07-20 00:00:00   
...                     ...                  ...   
3244073 2023-06-29 00:00:00  2023-06-29 04:00:00   
3244074 2023-05-31 00:00:00  2023-05-31 04:00:00   
3244075 2023-05-15 00:00:00  2023-05-15 04:00:00   
3244076 2023-05-15 00:00:00  2023-05-15 04:00:00   
3244077 2023-05-15 00:00:00  2023-05-15 04:00:00   

                         keyDevEventTypeName  
0                   M&A Transaction Closings  
1              M&A Transaction Announcements  
2                              Special Calls  
3                         Dividend Increases  
4                      Earnings Release Date  
...                                      ...  
3244073         Bankruptcy – Reorganization   
3244074  Bankruptcy – Asset Sale/Liquidation  
3244075                   Bankruptcy - Other  
3244076                  Bankruptcy - Filing  
3244077                   Bankruptcy - Other  

[3244074 rows x 8 columns]

In [17]:
# 'companyName' 컬럼 안에 있는 고유한 데이터 확인
unique_values = keydev['companyName'].unique()
num_unique = len(unique_values)
print("컬럼 'Category' 안에 몇 종류의 데이터가 있는지:", num_unique)
print("고유한 데이터:", unique_values)

컬럼 'Category' 안에 몇 종류의 데이터가 있는지: 454483
고유한 데이터: ['21 Invest Sgr S.p.A.' '3i Group plc' 'Pappas Capital, LLC' ...
 'Aries G.K.'
 'Industrial manufacturing property located in Alleyton, Texas'
 'Centennial Emergency Physicians, LLC']


In [19]:
from sklearn.preprocessing import LabelEncoder
label_encoder = LabelEncoder()
keydev['companyName'] = label_encoder.fit_transform(keydev['companyName'])

In [21]:
keydev

companyId                           companyName      keyDevId  \
0        1.850100e+04                  21 Invest Sgr S.p.A.  1.786392e+09   
1        1.850100e+04                  21 Invest Sgr S.p.A.  1.775873e+09   
2        1.851100e+04                          3i Group plc  1.856147e+09   
3        1.851100e+04                          3i Group plc  1.846370e+09   
4        1.851100e+04                          3i Group plc  1.843429e+09   
...               ...                                   ...           ...   
3244069  2.147481e+09  Centennial Emergency Physicians, LLC  1.846943e+09   
3244070  2.147481e+09  Centennial Emergency Physicians, LLC  1.846939e+09   
3244071  2.147481e+09  Centennial Emergency Physicians, LLC  1.838307e+09   
3244072  2.147481e+09  Centennial Emergency Physicians, LLC  1.837743e+09   
3244073  2.147481e+09  Centennial Emergency Physicians, LLC  1.838307e+09   

                                                  headline  \
0        A&M Capital Europe acquired Carton Pack Srl fr...   
1        A&M Capital Europe agreed to acquire Carton Pa...   
2                              3i Group plc - Special Call   
3                           3i Group plc Approves Dividend   
4        3i Group plc to Report Q1, 2024 Results on Jul...   
...                                                    ...   
3244069  Joint Reorganization Plan and Disclosure State...   
3244070  Motion for De Minimis Asset Sale Filed by Envi...   
3244071  Motion for Joint Administration Approved for E...   
3244072  Envision Healthcare Corporation Filed for Bank...   
3244073  Motion for Joint Administration Filed by Envis...   

                                                 situation  \
0        A&M Capital Europe agreed to acquire Carton Pa...   
1        A&M Capital Europe agreed to acquire Carton Pa...   
2                              3i Group plc - Special Call   
3        3i Group plc at its Annual General Meeting hel...   
4        3i Group plc announced that they will report Q...   
...                                                    ...   
3244069  Envision Healthcare Corporation, along with it...   
3244070  Envision Healthcare Corporation filed a motion...   
3244071  The US Bankruptcy Court granted an order for t...   
3244072  Envision Healthcare Corporation, along with it...   
3244073  Envision Healthcare Corporation, along with it...   

               announcedDate mostImportantDateUTC  \
0        2022-04-21 00:00:00  2022-04-21 00:00:00   
1        2022-03-11 00:00:00  2022-03-11 00:00:00   
2        2023-09-20 06:00:00  2023-09-20 13:00:00   
3        2023-06-29 14:30:00  2023-06-29 14:30:00   
4        2023-06-16 00:00:00  2023-07-20 00:00:00   
...                      ...                  ...   
3244069  2023-06-29 00:00:00  2023-06-29 04:00:00   
3244070  2023-05-31 00:00:00  2023-05-31 04:00:00   
3244071  2023-05-15 00:00:00  2023-05-15 04:00:00   
3244072  2023-05-15 00:00:00  2023-05-15 04:00:00   
3244073  2023-05-15 00:00:00  2023-05-15 04:00:00   

                         keyDevEventTypeName  
0                   M&A Transaction Closings  
1              M&A Transaction Announcements  
2                              Special Calls  
3                         Dividend Increases  
4                      Earnings Release Date  
...                                      ...  
3244069         Bankruptcy – Reorganization   
3244070  Bankruptcy – Asset Sale/Liquidation  
3244071                   Bankruptcy - Other  
3244072                  Bankruptcy - Filing  
3244073                   Bankruptcy - Other  

[3244074 rows x 8 columns]

keydev 전처리 관련 질문
1. 텍스트 분석(감성분석)을 어떻게 해야할 지 -> situation 컬럼 하나로 하자
    의문 : 감성분석할 때, (+)(-)(0) 를 구별해야하는데, 
    감성 분석 하려면, 컬럼(situation) 사용보다 뉴스 크롤링해야하는데, 얘가 메인업무량이 될까바 ,,, 흠 
2. 회사 종류 454483개 -> 얘를 어떻게 해야할 지 : 상장/비상장인지 여부를 여기서 확인하거나 다른 df 연동시켜야하는데 연동방법 몰라,, 
3. companyID -> company table에서 ticker와 연동하면되고, 구글링 ㄱ

의견
- 최대한 간단하게 작업을 끝내놓고(산업군 분석, 상승할/하락할 산업군 비중 최적화 한다던지)
- 거시 경제 지표 최대한 활용해서 산업군 분석 예정(금리, 환율, 인플레이션, 경제 성장률, 고용률, 실업률 등등 이 중에서 있는 데이터 활용)
- 산업군 분석 -> 상승할 것들 종목 확인 해서 가치지표 (PER, ROE) 활용해서 스크리닝한다던지 등
- 이렇게 할거면, 데이터 다시 봐서 산업군 분석에 사용할 수 있는 데이터 먼저 고르기 -> 가치지표 어떻게 활용할지 고르기
- 산업군 분석 어떻게 진행하는 지 조사 후 우리 데이터랑 쓸만한 컬럼 비교해서 의견 나누기
- 데이터 전처리 팁 ! 우선, 투자 안할 회사들(티커 company 테이블, 재균오빠 파일 참고) , 비상장종목, ticker-companyID 안맞는거 제거하면 용량 많이 제거됨

월요일 13시 30분 회의
- 지금까지 의견 나온거 종합해서 각자의 아이디어 활용해서 데이터 전처리 방안, 분석 방안, 구현 방안 논의할거 정리해오기 
- 월요일에 교수님 피드백 받을 내용 준비하기

In [1]:
print('dsfsfsdf')

dsfsfsdf
